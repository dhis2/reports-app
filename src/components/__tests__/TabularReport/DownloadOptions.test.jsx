import { shallow } from 'enzyme'
import React from 'react'
import {
    DownloadOptions,
    DownloadOption,
} from '../../TabularReport/DownloadOptions.jsx'

describe('<DownloadOptions />', () => {
    const fileUrls = [
        { extension: 'pdf', url: 'http://www.google.nl' },
        { extension: 'xls', url: 'http://www.google.com' },
    ]

    it('renders no <DownloadOption /> when fileUrls is empty', () => {
        const wrapper = shallow(<DownloadOptions fileUrls={[]} />)
        expect(wrapper.find(DownloadOption)).toHaveLength(0)
    })

    it('renders one <DownloadOption /> per file', () => {
        const wrapper = shallow(<DownloadOptions fileUrls={fileUrls} />)
        expect(wrapper.find(DownloadOption)).toHaveLength(fileUrls.length)
    })
})

describe('<DownloadOption />', () => {
    it('opens PDF files in a new window', () => {
        const file = { extension: 'pdf', url: 'http://www.google.nl' }
        const wrapper = shallow(<DownloadOption file={file} />)
        expect(wrapper.find('a').prop('target')).toBe('_blank')
    })

    it('opens non-PDF files in the same window', () => {
        const file = { extension: 'xls', url: 'http://www.google.com' }
        const wrapper = shallow(<DownloadOption file={file} />)
        expect(wrapper.find('a').prop('target')).toBe('_self')
    })

    it('renders the correct text for the link', () => {
        const file = { extension: 'pdf', url: 'http://www.google.nl' }
        const wrapper = shallow(<DownloadOption file={file} />)
        expect(wrapper.text().toLowerCase()).toContain('download as pdf')
    })

    it('renders the correct href', () => {
        const file = { extension: 'xls', url: 'http://www.google.com' }
        const wrapper = shallow(<DownloadOption file={file} />)
        expect(wrapper.find('a').prop('href')).toBe(file.url)
    })
})
