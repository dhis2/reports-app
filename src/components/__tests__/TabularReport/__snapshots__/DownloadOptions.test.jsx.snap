// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DownloadOptions/> Renders an empty container when fileUrls is an empty array 1`] = `
<div
  id="download-options-container"
>
  <style>
    
            div {
                display: flex;
                justify-content: flex-end;
            }
        
  </style>
</div>
`;

exports[`<DownloadOptions/> Renders an empty container when fileUrls is an empty array 2`] = `
<div
  id="download-options-container"
>
  <DownloadOption
    file={
      Object {
        "extension": "pdf",
        "url": "http://www.google.nl",
      }
    }
    key="0"
  />
  <DownloadOption
    file={
      Object {
        "extension": "xls",
        "url": "http://www.google.com",
      }
    }
    key="1"
  />
  <style>
    
            div {
                display: flex;
                justify-content: flex-end;
            }
        
  </style>
</div>
`;
